### PR TITLE
build: bump up precommit hooks

### PR DESCRIPTION
**Pull request by knyazer**
Also added ```toml``` filetype to ruff scanning, since ruff has some checks for the pyproject.toml structure: [RUF200](https://docs.astral.sh/ruff/rules/invalid-pyproject-toml/)

### CODE REVIEW

**Comment by knyazer** [comment]
Do you think it makes sense to wait for https://github.com/google/jax/pull/23492 to get merged, so that we can use one less ```# pyright: ignore```?

File: equinox/nn/_conv.py
Comment (patrick-kidger) [inline equinox/nn/_conv.py:1]: Are the changes here coming from RUFF? If so, why?
(I think I mildly prefer the earlier implementation -- with only one `return` I think the control flow is easier to understand / easier to check there's no accidental fallthrough behaviour.)

**Review by patrick-kidger — COMMENTED** [review]
No strong feelings about the pyright-ignores, happy to merge without that.

File: equinox/nn/_conv.py
Comment (knyazer) [reply equinox/nn/_conv.py:1]: Oops, actually this is an artefact from me trying to come up with a way to fix pyright type inference here... Sorry :)

**Comment by knyazer** [comment]
Fixed the typos, while I'm on it :wink:

**Comment by patrick-kidger** [comment]
This LGTM! Thank you for all of the (many, many) fixes.

Gosh, I like writing 'argments' instead of 'arguments'.

**Pull request merged**